teuthology/openstack/__init__.py

```diff
@@ -113,8 +113,16 @@ def get_volumes(self):
         Return the uuid of the volumes attached to the name_or_id
         OpenStack instance.
         """
-        volumes = self['os-extended-volumes:volumes_attached']
-        return [volume['id'] for volume in volumes ]
+        if 'os-extended-volumes:volumes_attached' in self.info:
+            volumes = self['os-extended-volumes:volumes_attached']
```

Comment:
This is weird - the `if` clause checks for the existence of `self.info['os-extended-volumes:volumes_attached']` but then it does `volumes = self['os-extended-volumes:volumes_attached']` ??

Reply:
In other words, should the `if` clause be `if 'os-extended-volumes:volumes_attached' in self:` ?

Reply:
The index operator is reading data from self.info, but it does not support 'in' operator, so it means `'os-extended-volumes:volumes_attached' in self` will not check the presence in `self.info`.